### ISSUE TITLE
Spike: Add more licenses using multiple license feature

### ISSUE THREAD

**Issue by jggautier** [IQSS/dataverse.harvard.edu#193]
The Harvard Dataverse Repository's license list includes only CC0. We'd like to add more licenses.

During a curation team meeting we decided to add the Creative Commons licenses listed at https://creativecommons.org/about/cclicenses.

To figure out what other licenses to add, we'd like to learn:
- What are other Dataverse installations using?
- Among deposits in Harvard's repo, what licenses have been entered in Terms of Use field when CC0 isn't used.

Steps for adding and editing licenses is at https://guides.dataverse.org/en/latest/installation/config.html#configuring-licenses

**Comment by jggautier** [comment]
**What are other Dataverse installations using?**
At least 36 Dataverse installations have the multiple license update (running v5.10 or later). A list of those installation's licenses is in the Google Sheet at https://docs.google.com/spreadsheets/d/1v5a1M3sQdVyNS0aeabm34MFHJmmaQ8JANIaxpNMlGrg

**Comment by jggautier** [comment]
Stafano recommended reviewing the licenses supported by repositories listed in the comparison chart at https://zenodo.org/record/7189481#.Y1GfJOzMI8Z.

In the research I've described in my earlier comments in this issue, I've looked into many of these licenses, but there are some that I haven't heard of before and haven't seen yet, e.g. Artistic 2.0 from OSF. So I'll look at the nature of those licenses and if datasets in the Harvard repo has used them.

**Comment by jggautier** [comment]
The licenses listed on page 2 of the comparison chart at https://zenodo.org/record/7189481#.Y1KmfezMI8a that I haven't already reviewed are the CERN OHL and TAPR OHL licenses, Artistic 2.0, and the LGPL licenses. All of these licenses are meant to be applied to software.

I've added this review to this issue's first comment so everything's in one place.

**Comment by mreekie** [comment]
@jggautier in reviewing this with Stefano, we had the question of what the intended outcome is here.
Is what's needed here a discussion of which licenses to implement?
If this is right,  Dd you know who the key players are ? If not we can figure it out.

**Comment by mreekie** [comment]
Technically this is likely easy. The work here is deciding which licensing.
This is not a dev thing.
Stefano, sonia, and julian need to decide.

**Comment by mreekie** [comment]
Confirmatin:
- Licenses are driven by json files. No code

**Comment by mreekie** [comment]
Julian
- Created a document that describes how we got to where we are.,
- There is an effort in the community to get to a standard list of license.
-  Curation team is meeting this week.
- Discussions of maintenance related work needs Caitlyn on board.
- This is time bounded to the upcoming sprint

**Comment by sbarbosadataverse** [comment]
@mreekie yes apparently into January at this point. Julian starts vacation next week, and I start the following week.

**Comment by mreekie** [comment]
Held over into Dec 15 sprint

**Comment by mreekie** [comment]
Sonia is out this week. 
Will definitely slip into next week, possibly into next sprint.

**Comment by mreekie** [comment]
Julian/Phil discussed PR #9262

**Comment by pdurbin** [comment]
Right, I think @qqmyers and I agree that as long as @jggautier sticks to licenses that are in the guides (instead of cooking up new JSON files) he shouldn't be blocked on waiting for this PR:

- https://github.com/IQSS/dataverse/pull/9262

The reason is that we'll need to migrate licenses for other installations to the new database schema anyway. So there's no need to wait.

**Comment by jggautier** [comment]
Thanks. Just want to clarify that "as long as @jggautier sticks to licenses that are in guides" should really be "as long as the Harvard Dataverse Repository sticks to licenses that are in the guides...". I don't want to give the impression that I'm making the final decision about which licenses to use or even that my opinion weighs more.

Lastly, I am recommending in this issue that Harvard's repo consider using some widely-used licenses that are not in the list at [#8512](https://github.com/IQSS/dataverse/pull/9262) as of this writing. But it looks to me like that should be okay as long as we follow the principles of:
- replacing spaces with dashes for what's put in the rightsIdentifier key, such as CC0-1.0 instead of CC0 1.0
- using SPDX's URLs in the schemeURI key (currently called the uri key), such as https://spdx.org/licenses/ instead of 
http://creativecommons.org/publicdomain/zero/1.0

**Comment by pdurbin** [comment]
@jggautier can we add JSON files to https://github.com/IQSS/dataverse/pull/9262 to keep Harvard Dataverse in sync with licenses what will be offered in the guides? Sounds like they're widely used and not Harvard-specific.

**Comment by jggautier** [comment]
Wouldn't that make https://github.com/IQSS/dataverse/pull/9262 dependent on this issue?

And isn't https://github.com/IQSS/dataverse/pull/9262 more about standardizing how licenses are represented in the metadata exports, and less about standardizing which licenses are used?

**Comment by jggautier** [comment]
An update about the status of this. The licenses to choose will be discussed during the next curation team meeting on Feb. 9.

I'm not clear enough about all of the changes I see in https://github.com/IQSS/dataverse/pull/9262. But I think @pdurbin suggested in a chat we had last week that once the curation team chooses the licenses:
- we use [the JSON files in the configuration guides](https://guides.dataverse.org/en/latest/installation/config.html#adding-creative-common-licenses) for whichever Creative Commons licenses are chosen
- we create JSON files for any licenses that aren't in that list in the guides

So we won't follow [the two principles I wrote about last week](https://github.com/IQSS/dataverse.harvard.edu/issues/193#issuecomment-1372560833), and if or when changes to the JSON files are made (that is, the way that each license is represented), the Harvard repo will need to make those changes.

During a meeting last year the curation team talked about how we can always add more licenses as needed, and I'd like to plan for how to figure out what's needed. Curators and depositors might ask us to add a license to the dropdown menu, but I also think we could plan to actively monitor the use of those licenses and what people are still typing in the Terms fields.

**Comment by jggautier** [comment]
This wasn't discussed on Feb. 9. @sbarbosadataverse's going to schedule a time for the curation team to meet about this.

**Comment by jggautier** [comment]
@sbarbosadataverse and I met today to discuss this and Sonia is scheduling a meeting with others on the curation team to discuss more.

Here's what we discussed today:
- We talked about licenses meant only for software. Sonia agreed those licenses should be excluded for now.
- We talked about how it would be helpful to know how often the standard data licenses have been used. So for each license we're considering (the different versions of the 6 CC licenses, the Open Commons license and the Open Government license), I'll get a count of the datasets with those licenses applied (considering only the latest published version of each dataset) and add that info in another comment in this GitHub issue.
- We talked about license versions. Sonia would like the dropdown menu to list only the most recent versions of each license. And we would consider cases where many datasets have used an older version of a license. We could ask the depositor if they can update the datasets to use the newest version of the license. If that's not possible, we could add that older version of the license to the dropdown menu, then the depositor can update their datasets.

  I just noticed that the Digital Curation Centre's [data license guide](https://www.dcc.ac.uk/guidance/how-guides/license-research-data#x1-8000) talks about the differences between the versions of the CC licenses. Looks like each version has been more permissive and less restrictive than the previous version.
- We talked about plans to update existing datasets with standard licenses (e.g. CC licenses) in the Terms fields, so that those licenses are chosen from the dropdown menu, making them more machine-readable and findable. Sonia talked about encouraging depositors to update their datasets by:
    - Emailing depositors
    - Adding information on the support website (https://support.dataverse.harvard.edu) about multiple license support
    - Tweeting about multiple license support
-  We talked about the experience of choosing licenses from the existing dropdown menu. The dropdown menu handles flat lists. That component is okay for now since we don't think the list of licenses being added to the Harvard Dataverse will be that long, but if/when the list gets longer another component would make it easier for depositors and curators to find and choose licenses. This improvement could be suggested and tracked in another GitHub issue, possibly in the main Dataverse GitHub repo. The last time I checked, there were two Dataverse repositories have over 20 licenses in their dropdown menu.

**Comment by qqmyers** [comment]
FWIW: The license mechanism distinguishes between which are installed and which of those are active/show up in the edit menu as options. Hopefully that handles the versioning bullet above.

**Comment by jggautier** [comment]
Thanks, yeah that seems helpful if installation admins want to update datasets that have a previous versions of a license and then "deactivate" those previous versions so that they can't be applied to other datasets (unless we "activate" them again).

What I wasn't sure about is if we should keep those older versions of the licenses active, so depositors can continue to apply them, either when updating their datasets to take advantage of the dropdown or when adding or migrating new datasets. But looking at the count of CC licenses used, most use 4.0 versions and relatively very few use older, pre-4.0 versions.

I wrote yesterday that I'd get a count of datasets using each CC license, so here's an estimate*:
<img width="520" alt="Screen Shot 2023-02-16 at 12 18 13 PM" src="https://user-images.githubusercontent.com/18374574/219439783-cded2fa6-eec3-4df4-9a8d-ae47ec9911de.png">

* This is from metadata collected in October 2022. And to simplify things, the counts are based only on URLs of CC licenses. So it doesn't count the cases where the depositor didn't enter the URL of the license in a Terms field, such as the dataset at https://doi.org/10.7910/DVN/0BEZK5. But of the roughly 32,000 datasets with "custom terms", about 27,000 datasets have  URLs of CC licenses in their Terms fields.

**Comment by jggautier** [comment]
I added the 10 new licenses to the Harvard Dataverse. Also added them to the Demo Dataverse.

I'll open a new issue in the Dataverse GitHub repo about adding to the Dataverse Guides the JSON files for the four non-Creative Commons licenses that aren't there as of this writing.

**Comment by jggautier** [comment]
Oh, I started collecting the license and terms metadata of published datasets in the Harvard Dataverse. They're in [a Google Sheet](https://docs.google.com/spreadsheets/d/1keYomqTbJ6Yu8EkEtlrezYixR2BG4WbWKlfGsUeWe-8). I'm working on updating that info on a regular basis and adding other views so we can see the metadata before and after the addition of these licenses, how often each license is being used, and which licenses people continue to use that aren't in the dropdown list.

**Comment by jggautier** [comment]
I wrote that in May 2023 I'd look at how people were applying licenses and terms metadata since the new licenses were added in Harvard Dataverse. I'm very late, but here's a Google Sheet with the 8378 datasets that been published since the new licenses were added in late February 2023:

https://docs.google.com/spreadsheets/d/1keYomqTbJ6Yu8EkEtlrezYixR2BG4WbWKlfGsUeWe-8

The spreadsheet's first tab shows the counts of datasets that use each license or use no license.

I haven't looked closely at this yet, but last year I do remember seeing these cases:
- Depositors published datasets with a "custom license" and used the Terms of Use field to enter a license that's available in the License/Data Use Agreement dropdown menu. For example, see the dataset at https://doi.org/10.7910/DVN/84NLSQ and many of its collection's other datasets. I wonder if the admins of that collection are publishing datasets using the APIs and aren't aware of the multiple license dropdown in the UI.
- Depositors published datasets before they could choose licenses from a dropdown list, so they entered standard license info, like the URL of the CCBY license, in the Terms of Use field. Then they published a new version but didn't choose CCBY from the License/Data Use Agreement dropdown menu. I'd guess this is because they didn't know that they could or should choose the same license from the new dropdown menu.
- Depositors published datasets where they entered a standard license in the Terms of Use field because they wanted to use other metadata fields, like the Disclaimer field, that become available only if you choose "Custom License. There's been discussion among the community about what could or should be done to support this need while ensuring that the license metadata is as machine-readable as it would be if they chose the same license from the License/Data Use Agreement dropdown menu.

I also haven't looked closely at cases where depositors have used the Terms of Use fields to enter standard licenses that we might consider adding to the License/Data Use Agreement dropdown menu.

Lastly, after the MIT license is added to the dropdown list as part of https://github.com/IQSS/dataverse.harvard.edu/issues/248, I can update the spreadsheet again, hopefully sooner this time!, so that we can see how that license's been used.
